internal/mining/mining.go

```diff
@@ -1040,9 +1040,10 @@ func (g *BlkTmplGenerator) addAutoRevocationsToQueue(winningTickets map[chainhas
 
 	// Create revocation transactions for all tickets that will become missed
 	// or expired as of this block.
-	for _, ticketHash := range revokeTickets {
+	for i := range revokeTickets {
 		// Create the revocation transaction.
-		txDesc, err := g.createRevocationFromTicket(&ticketHash, blockUtxos,
+		ticketHash := &revokeTickets[i]
+		txDesc, err := g.createRevocationFromTicket(ticketHash, blockUtxos,
```

Comment:
nit: Maybe consumers should decide if they need a pointer? Just watching out for consistency.

Reply:
I assume you mean within the loop, right?

Most of the time, it's better to use a pointer for the reference and then only dereference it if needed because it almost always avoids a copy.

Good catch on the lack of consistency though as I do see that the prior change (in `treasury.go`) is a copy and it really should've been a pointer itself to avoid that copy.  I'll update that here in a sec.

Reply:
> I assume you mean within the loop, right?

yh.

> Most of the time, it's better to use a pointer for the reference and then only dereference it if needed because it almost always avoids a copy.

Makes sense.